scirpy/io/_io.py

```diff
@@ -324,42 +324,126 @@ def read_airr(path: Union[str, Sequence[str], Path, Sequence[Path]]) -> AnnData:
     """
     ir_objs = {}
 
-    if isinstance(path, str) or isinstance(path, Path):
+    if isinstance(path, str) or isinstance(path, Path) or isinstance(path, pd.DataFrame):
```

Comment:
```suggestion
    if isinstance(path, (str, Path, pd.DataFrame)):
```

(just found out myself that this is possible)